### ISSUE TITLE
Plenoxels / Spherical Harmonics Raymarcher

### ISSUE THREAD

**Issue by d4l3k** [facebookresearch/pytorch3d#1435]
## 🚀 Feature

NOTE: Please look at the existing list of Issues tagged with the label ['enhancement`](https://github.com/facebookresearch/pytorch3d/issues?q=label%3Aenhancement). **Only open a new issue if you do not see your feature request there**.

## Motivation

Plenoxels (https://arxiv.org/pdf/2112.05131.pdf) use spherical harmonics to avoid having to use a MLP to generate the color outputs. It'd be great to have a raymarcher that could render using spherical harmonics to improve training speed as well as make it easier to use SH in other tasks.

It's not terribly feasible to support the sparse voxels/octrees used by plenoxels or plenoctrees but having support for dense SH models would be very helpful for projects like https://github.com/d4l3k/torchdrive

Sample implementation with sparse voxels in CUDA (we'd only need to implement the SH part though): https://github.com/sxyu/svox2/blob/master/svox2/csrc/render_lerp_kernel_nvol.cu

Related #695 issue to use spherical harmonics for lighting

cc @bottler
